2021-04-22 16:09:13,949 ERROR: Exception on /reset_password_request [POST] [in e:\flask\microblog\venv\lib\site-packages\flask\app.py:1892]
Traceback (most recent call last):
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "e:\flask\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "E:\flask\microblog\app\routes.py", line 168, in reset_password_request
    send_password_reset_email(user)
  File "E:\flask\microblog\app\email.py", line 20, in send_password_reset_email
    token = user.get_reset_password_token()
  File "E:\flask\microblog\app\models.py", line 67, in get_reset_password_token
    return jwt.encode({'reset_password': self.id, 'exp': time() + expires_in}, app.config['SECRET_KEY'], algorithm='HS256').decode('utf-8')
AttributeError: 'str' object has no attribute 'decode'
2021-04-22 16:12:23,515 INFO: Microblog startup [in E:\flask\microblog\app\__init__.py:52]
2021-04-22 16:12:28,349 ERROR: Exception on /reset_password_request [POST] [in e:\flask\microblog\venv\lib\site-packages\flask\app.py:1892]
Traceback (most recent call last):
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "e:\flask\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "E:\flask\microblog\app\routes.py", line 168, in reset_password_request
    send_password_reset_email(user)
  File "E:\flask\microblog\app\email.py", line 20, in send_password_reset_email
    token = user.get_reset_password_token()
  File "E:\flask\microblog\app\models.py", line 72, in get_reset_password_token
    return jwt.encode({'reset_password': self.id, 'exp': time() + expires_in},app.config['SECRET_KEY'], algorithm='HS256').decode('utf-8')
AttributeError: 'str' object has no attribute 'decode'
2021-04-22 16:15:50,139 INFO: Microblog startup [in E:\flask\microblog\app\__init__.py:52]
2021-04-22 16:15:58,480 ERROR: Exception on /reset_password_request [POST] [in e:\flask\microblog\venv\lib\site-packages\flask\app.py:1892]
Traceback (most recent call last):
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "e:\flask\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "e:\flask\microblog\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "E:\flask\microblog\app\routes.py", line 168, in reset_password_request
    send_password_reset_email(user)
  File "E:\flask\microblog\app\email.py", line 20, in send_password_reset_email
    token = user.get_reset_password_token()
  File "E:\flask\microblog\app\models.py", line 72, in get_reset_password_token
    return encoded_jwt.decode('UTF-8')
AttributeError: 'str' object has no attribute 'decode'
2021-04-22 16:17:40,288 INFO: Microblog startup [in E:\flask\microblog\app\__init__.py:52]
2021-04-22 16:22:17,759 INFO: Microblog startup [in E:\flask\microblog\app\__init__.py:52]
2021-04-23 22:30:39,775 INFO: Microblog startup [in E:\flask\microblog\app\__init__.py:54]
2021-04-23 22:52:57,101 INFO: Microblog startup [in E:\flask\microblog\app\__init__.py:56]
